python/lsst/analysis/tools/tasks/isrTaskDetectorExposureMetricsAnalysis.py

```diff
@@ -0,0 +1,66 @@
+# This file is part of analysis_tools.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ("IsrDetectorExposureMetricsAnalysisConfig", "IsrAmpOffsetMetricsAnalysisTask")
+
+import pandas as pd
+from lsst.pipe.base import NoWorkFound, connectionTypes
+
+from ..interfaces import AnalysisBaseConfig, AnalysisBaseConnections, AnalysisPipelineTask
+
+
+class IsrDetectorExposureMetricsAnalysisConnections(
+    AnalysisBaseConnections,
+    dimensions=("instrument", "exposure", "detector"),
+):
+    metadata = connectionTypes.Input(
+        doc="Task metadata from ISR",
+        name="isr_metadata",
+        storageClass="TaskMetadata",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+
+
+class IsrDetectorExposureMetricsAnalysisConfig(
+    AnalysisBaseConfig, pipelineConnections=IsrDetectorExposureMetricsAnalysisConnections
+):
+    pass
+
+
+class IsrAmpOffsetMetricsAnalysisTask(AnalysisPipelineTask):
+    ConfigClass = IsrDetectorExposureMetricsAnalysisConfig
+    _DefaultName = "isrAmpOffsetMetricsAnalysis"
+    _SubTaskName = "ampOffset"
```

Comment:
What is SubTaskName here for?

Reply:
I think this is not the right place for this. Tasks should be specialized to their input dataset type, which in this case is isr metadata, not for the values inside the dataset, that is something for the analysis tools to care about. I think you need to orthogonalize these concerns

Reply:
`subTaskName` is now a config `Field` within `IsrMetadataExposureDetectorAnalysisConfig`. Does this satisfy the concerns of @natelust? I note that not supplying a `subTaskName` will default to `None`, which would allow an analysis tool to handle it.

Reply:
I believe it does to a considerable extent.

Reply:
Do we need this config field at all? I.e., what's the benefit of selecting a subtask here, rather than passing the entire metadata through to an analysis tool to subselect on itself?

Reply:
Agreed, we should be able to pass the entire metadata to an analysis tool for subselection. We just need a more sophisticated action to locate the data. I'll test it out.

Reply:
Done! Needed some refactoring.